src/ArrayUtils.f90

```diff
@@ -991,4 +995,69 @@ PURE FUNCTION findUpBound_1DReal(r,pos,delta,incl,tol) RESULT(val)
   ENDIF
 ENDFUNCTION findUpBound_1DReal
 !
+!-------------------------------------------------------------------------------
+!> @brief Routine that checks that the entries of r either monotonically increase or decrease.
+!>        Note this is limited to 1 dimensional real arrays.
+!> @param r input array to check for monotonicity of values
+!> @returns good boolean indicating whether the check has passed or not
+!>
+FUNCTION Is_Mono(r) RESULT(good)
+  REAL(SRK),INTENT(IN) :: r(:)
+  LOGICAL(SBK) :: good
+
+  IF(SIZE(r) > 1) THEN
+    IF(r(1) < r(2)) THEN
+      good=Is_Mono_Increasing(r)
+    ELSE
+      good=Is_Mono_Decreasing(r)
+    ENDIF
+  ELSE
+    good=.FALSE.
+  ENDIF
+ENDFUNCTION Is_Mono
+!
+!-------------------------------------------------------------------------------
+!> @brief Routine that checks that the entries of r either monotonically increase or decrease.
+!>        Note this is limited to 1 dimensional real arrays.
+!> @param r input array to check for monotonicity of values
+!> @returns good boolean indicating whether the check has passed or not
+!>
+FUNCTION Is_Mono_Increasing(r) RESULT(good)
+  REAL(SRK),INTENT(IN) :: r(:)
+  INTEGER(SIK) :: i
+  LOGICAL(SBK) :: good
+  good=.TRUE.
+
+  IF(SIZE(r) > 1) THEN
+    DO i=2,SIZE(r)
+      IF(r(i) < r(i-1)) good=.FALSE.
+    ENDDO
+  ELSE
+    good=.FALSE.
```

Comment:
should this return true if the size is 1?

Reply:
@aarograh, @HendersonSC, and I had a discussion on this. Our general consensus was if something is size 1 it is impossible to be increasing, decreasing, or generally monotonic and it would be likely that any routine that is relying on monotonic behavior probably requires more than a single scalar entry, hence being false.